WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -76,9 +76,49 @@ private void StepInputRegistrationPhase()
 				}
 				else
 				{
-					round.SetPhase(Phase.ConnectionConfirmation);
+					var areAllTxoUnspent = await CheckTxoSpendStatusAsync(round, cancel).ConfigureAwait(false);
+					if (areAllTxoUnspent)
+					{
+						round.SetPhase(Phase.ConnectionConfirmation);
+					}
+					else
+					{
+						round.SetPhase(Phase.Ended);
+						round.LogInfo($"The round was disrupted by users who spent their coins.");
+					}
+				}
+			}
+		}
+
+		private async Task<bool> CheckTxoSpendStatusAsync(Round round, CancellationToken cancellationToken)
+		{
+			IEnumerable<(Alice Alice, Task<GetTxOutResponse?> Status)> GetTxoSpendingStatus()
+			{
+				foreach (var alice in round.Alices)
+				{
+					var input = alice.Coin.Outpoint;
+					var getTxOutTask = Rpc.GetTxOutAsync(input.Hash, (int)input.N, includeMempool: true);
+					yield return (alice, getTxOutTask);
+				}
+			}
+
+			var allTxoAreUnspent = true;
+			var batchedRpc = Rpc.PrepareBatch();
+			var spendStatusCheckingTasks = GetTxoSpendingStatus().ToArray();
+
+			cancellationToken.ThrowIfCancellationRequested();
```

Comment:
I think that this line is not really useful because you get here "immediately".

Reply:
That's the point. I want to be able to cancel this _before_ sending and waiting for a huge batch of non-cancellable work to the bitcoin node.

Reply:
My point was: You must be *very lucky* to cancel actually.

Reply:
IMO it doesn't really matter either way, because cancellation tokens were be added to NBitcoin (already merged, but not released) and that will eventually supersede this.

Reply:
It makes more sense now, i believe.